File: app/src/main/java/hu/bme/szn/randomjokes/ui/newjokes/NewJokesPresenter.java
package hu.bme.szn.randomjokes.ui.newjokes;

import javax.inject.Inject;

import hu.bme.szn.randomjokes.model.Joke;
import hu.bme.szn.randomjokes.ui.Presenter;

public class NewJokesPresenter extends Presenter<NewJokesScreen> {

    @Inject
    public NewJokesPresenter() {
    }

    @Override
    public void attachScreen(NewJokesScreen screen) {
        super.attachScreen(screen);
    }

    @Override
    public void detachScreen() {
        super.detachScreen();
    }


    public void addNewJoke(
            Joke joke) {
        screen.addNewJoke(joke);
    }
}

File: app/src/main/java/hu/bme/szn/randomjokes/ui/newjokes/NewJokesPresenter.kt
package hu.bme.szn.randomjokes.ui.newjokes

import hu.bme.szn.randomjokes.model.Joke
import hu.bme.szn.randomjokes.ui.Presenter


class NewJokesPresenter : Presenter<NewJokesScreen>() {

    override fun attachScreen(screen: NewJokesScreen) {
        super.attachScreen(screen)
        //TODO
    }

    override fun detachScreen() {
        super.detachScreen()
        //TODO
    }

    fun addNewJoke(
            joke: Joke) {
        screen?.addNewJoke(joke)
    }

}